defrag/modules/helpers/services_manager.py

```diff
@@ -64,37 +65,121 @@ def __setitem__(self, key: str, item: Service) -> None:
         else:
             raise Exception(
                 f"Cannot add a service twice, yet you tried to add {key}")
-    
+
     def __getitem__(self, key: str) -> Service:
         if not key in self.data:
             raise Exception(f"Tried to access a nonexistent service: {key}")
         else:
             return super().__getitem__(key)
 
+    def list_all(self) -> List[str]:
+        return list(self.data.keys())
+
+    def list_on(self, predicate: Callable) -> List[str]:
+        return [s.name for s in self.values() if predicate(s)]
+
 
 class ServicesManager:
 
     services: Optional[Services] = None
-    services_names = ["twitter_reddit", "bugzilla", "wikis", "mailing_lists",
-                      "matrix", "telegram", "pagure", "zypper", "opi", "people", "activities"]
+    """ Using the structure below instead of shoving everything to class constructors to
+    have a 'dashboard view' of all defaults for all services. """
+    services_templates_to_services = {
```

Comment:
We need to think about this, because it would be not nice if adding a module would involve editing this file. I'd like it to be so that you just add a file to the `modules` folder and restart, and with that it gets loaded.

Reply:
I think this relates do your previous comment. I am not sure about the usage we are aiming for...